pkgs/development/compilers/llvm/common/default.nix

```diff
@@ -218,6 +220,13 @@ let
         extraBuildCommands = mkExtraBuildCommands cc;
       };
 
+      systemLibcxxClang = wrapCCWith rec {
+        cc = tools.clang-unwrapped;
+        libcxx = darwin.libcxx;
+        extraPackages = [ targetLlvmLibraries.compiler-rt ];
+        extraBuildCommands = mkExtraBuildCommands cc;
+      };
+
```

Comment:
Let's add a comment here to explain what this is useful for. I plan on documenting LLVM in nixpkgs so having a comment for reference will be helpful.

Reply:
Sure. It would be cool if this were something that could be controlled by https://github.com/NixOS/nixpkgs/pull/365057.

Reply:
It may be, the logic won't be implemented in that PR. However, that PR does form the basis. Another possibility is when I refactor `pkgs/development/compilers/llvm/common/default.nix`, a special function could be provided to easily create a LLVM/Clang CC wrapper.

Reply:
With all the different C++ and C++ ABI libraries, it would be nice if the boilerplate could be reduced and switching between them made easier.